### PR TITLE
Pr90x L1T stage2 for phase2 fix

### PR DESCRIPTION
**Pull request by rekovic**
This PR invokes Stage2 L1Trigger for Phase2.  It supersedes #17248 and #17462.  
It contains:
 * re-reverted #17248.
 *  add Stage2 products to the RawCollector in case of Phase2 era (#17462)
 * Silencing EMTF (MuonEndCapTrackFinder) cout statements.

Original #17248 problems due to HCAL TPs is now fixed (merged #17540).
Overflooding messages from EMTF are silenced as well.

For completion, below is the original PR info:

#17248 PR 90x for change in configuration to use L1T Stage2 (as opposed to L1T Legacy) in Phase2 era. This PR also includes (needed to pass tests) https://github.com/cms-sw/cmssw/pull/17260 which fixed crashes in HCAL TPs for Phase2 

This PR is needed for Phase2 MC Production.

Details:
  - Use phase2_common era modifier to 
    - switch in L1T configuration to use Stage2 
    - configure conditions of L1T Muon, Calorimeter, and Global (as in Stage2)
    - configure L1T RAW2DIGI and DIGI2RAW to use Stage2 products
  - Include back in  simHcalTrigPrimitiveDigis and simHcalTTPDigis into hcalDigiSequence

### CODE REVIEW

**Comment by rekovic** [comment]
please test

**Comment by rekovic** [comment]
+1

**Comment by civanch** [comment]
+1
DT differences for Phase2 are expected?

File: L1Trigger/Configuration/python/L1TDigiToRaw_cff.py [hunk 1]
```diff
@@ -15,11 +15,15 @@
...
+from Configuration.Eras.Modifier_phase2_common_cff import phase2_common
+phase2_common.toModify( rawDataCollector.RawCollectionList, func = lambda list: list.append(cms.InputTag("caloStage2Raw")) )
+phase2_common.toModify( rawDataCollector.RawCollectionList, func = lambda list: list.append(cms.InputTag("gmtStage2Raw")) )
+phase2_common.toModify( rawDataCollector.RawCollectionList, func = lambda list: list.append(cms.InputTag("gtStage2Raw")) )
 
 #
 # Legacy Trigger:
 #
-if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen()):
+if not (stage1L1Trigger.isChosen() or stage2L1Trigger.isChosen() or phase2_common.isChosen()):
```
Comment (davidlange6) [inline L1Trigger/Configuration/python/L1TDigiToRaw_cff.py]: Hi @rekovic - two issues
1) we agreed to clean up these isChosen..
2) you don't want to add phase2_common here, rather, you want to add what trigger you want into phase2_common era definition, Then this code is unchanged.

File: SimCalorimetry/Configuration/python/hcalDigiSequence_cff.py [hunk 2]
```diff
@@ -9,7 +9,8 @@
                                 +simHcalDigis
                                 *simHcalTTPDigis)
 
-_phase2_hcalDigiSequence = hcalDigiSequence.copyAndExclude([simHcalTriggerPrimitiveDigis,simHcalTTPDigis])
+#_phase2_hcalDigiSequence = hcalDigiSequence.copyAndExclude([simHcalTriggerPrimitiveDigis,simHcalTTPDigis])
```
Comment (davidlange6) [inline SimCalorimetry/Configuration/python/hcalDigiSequence_cff.py]: these lines can just be removed. Perhaps its worth a small PR with just this change as the first step?

**Comment by davidlange6** [comment]
-1
@rekovic and I have discussed the needed changes

File: SimCalorimetry/Configuration/python/hcalDigiSequence_cff.py (hunk 2 above)
Comment (rekovic) [inline SimCalorimetry/Configuration/python/hcalDigiSequence_cff.py]: Done. Now in #17628 .

File: L1Trigger/Configuration/python/L1TDigiToRaw_cff.py (hunk 1 above)
Comment (rekovic) [inline L1Trigger/Configuration/python/L1TDigiToRaw_cff.py]: Done.  Now in #17629.

**Comment by rekovic** [comment]
@civanch 
> DT differences for Phase2 are expected?

I did not understand your question. Could you clarify, please?

**Pull request closed without merging**